### PR TITLE
Fix bug in assertions returning true for some tests

### PR DESCRIPTION
**Pull request by galadim1**
**Previous issues**
Most of the existing tests were written incorrectly. 
The `assertions` are inside a promise which does not return. The `test` will always return true.

**Improvements Done**
1. All `tests` which does not return anything were rewritten
2. `Await/async` used instead of `done/done()` for deprecation error tests

**How to properly test**
To properly test;
1. Clone the repo
2. Change any assertion
3. Run the test
4. The test should fail e.g

For more context, here is the shortcut ticket
https://app.shortcut.com/chartmogul/story/57880/rewrite-the-assertion-in-chartmogul-node-library

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
-    "eslint-plugin-n": "",
-    "eslint-plugin-promise": "",
-    "eslint-plugin-standard": "",
-    "mocha": "",
-    "nock": "",
-    "nyc": "^15.1.0",
-    "pre-commit": ""
-  },
-  "pre-commit": "prepare"
+	"name": "chartmogul-node",
```
Comment (kamilpavlicko) [inline package.json:2]: is that necessary change, though?

**Review by whobubble — APPROVED** [review]
Test suite works (Tested locally) 
package.json formatting can be reverted

File: package.json (hunk 1 above)
Comment (galadim1) [reply package.json:2]: Sorry, my bad. My IDE has a different setup. i will revert the changes

Comment (galadim1) [reply package.json]: The problem was VSC prettier plugin indenting with` tabs` when I commit changes.

**Pull request merged**